src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -6525,6 +6619,12 @@ static const int64_t right_3_bits = right_n_bits(3);
       StubRoutines::_crc_table_adr = (address)StubRoutines::riscv::_crc_table;
       StubRoutines::_updateBytesCRC32 = generate_updateBytesCRC32();
     }
+
+    if (vmIntrinsics::is_intrinsic_available(vmIntrinsics::_float16ToFloat) &&
+        vmIntrinsics::is_intrinsic_available(vmIntrinsics::_floatToFloat16)) {
```

Comment:
Since this stub uses instructions from the Zfh extension which is not always available, do we need a similar checking like x86 [1] (https://bugs.openjdk.org/browse/JDK-8303415)? I see `vmIntrinsics::is_intrinsic_available` delegates work to `VM_Version::is_intrinsic_supported` [2].

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/vm_version_x86.cpp#L3333
[2] https://github.com/openjdk/jdk/blob/master/src/hotspot/share/classfile/vmIntrinsics.cpp#L671

Reply:
Thanks for catching!